great_expectations/expectations/metrics/table_metrics/table_column_types.py

```diff
@@ -87,6 +87,10 @@ def _spark(
         return spark_column_metadata
 
 
+def _get_normalized_table_name(table_name: str):
+    return table_name.strip("[]").strip('"')
```

Comment:
This seems MSSQL specific. I would rename this method to `_get_mssql_normalized_table_name` or something similar and check the dialect before calling via `execution_engine.dialect_name == GXSqlDialect.MSSQL`.

I am not familiar with MSSQL. Can square brackets or double quotes appear in table names if preceded by an escape character? That is, are there table names that will be broken with these `strip` calls?

Reply:
Fair point. Square brackets are allowed in MSSQL but then need to be escaped when running queries against the table. A better solution would be to let GX add the square brackets to the db object names when needed. Currently, I have to add them when adding a data asset, e.g. asset = datasource.add_table_asset(name="ClientCodes", schema_name="CR", table_name="[247_CLIENT_CODES]").